napari/layers/tracks/tracks.py

```diff
@@ -0,0 +1,520 @@
+        # self.events.data()
+        self._update_dims()
+
+    @property
+    def properties(self) -> Dict[str, np.ndarray]:
+        """ return the list of track properties """
+        return self._manager.properties
+
+    @property
+    def properties_to_color_by(self) -> List[str]:
+        """ track properties that can be used for coloring etc... """
+        return list(self.properties.keys())
+
+    @properties.setter
+    def properties(self, properties: Dict[str, np.ndarray]):
+        """ set track properties """
+        self._manager.properties = properties
+        self.events.properties()
+        self.events.color_by()
+
+    @property
+    def graph(self) -> list:
+        """ return the graph """
+        return self._manager.graph
+
+    @graph.setter
+    def graph(self, graph: dict):
+        self._manager.graph = graph
+        self._manager.build_graph()
+        self.events.rebuild_graph()
+
+    @property
+    def tail_width(self) -> Union[int, float]:
+        """float: Width for all vectors in pixels."""
+        return self._tail_width
+
+    @tail_width.setter
+    def tail_width(self, tail_width: Union[int, float]):
+        self._tail_width = tail_width
+        self.events.tail_width()
+        self.refresh()
+        self.status = format_float(self.tail_width)
+
+    @property
+    def tail_length(self) -> Union[int, float]:
+        """float: Width for all vectors in pixels."""
+        return self._tail_length
+
+    @tail_length.setter
+    def tail_length(self, tail_length: Union[int, float]):
+        self._tail_length = tail_length
+        self.events.tail_length()
+        self.refresh()
+        self.status = format_float(self.tail_length)
+
+    @property
+    def display_id(self) -> bool:
+        """ display the track id """
+        return self._display_id
+
+    @display_id.setter
+    def display_id(self, value: bool):
+        self._display_id = value
+        self.events.display_id()
+        self.refresh()
+
+    @property
+    def display_tail(self) -> bool:
+        """ display the track tail """
+        return self._display_tail
+
+    @display_tail.setter
+    def display_tail(self, value: bool):
+        self._display_tail = value
+        self.events.display_tail()
+        self.refresh()
+
+    @property
+    def display_graph(self) -> bool:
+        """ display the graph edges """
+        return self._display_graph
+
+    @display_graph.setter
+    def display_graph(self, value: bool):
+        self._display_graph = value
+        self.events.display_graph()
+        self.refresh()
+
+    @property
+    def color_by(self) -> str:
+        return self._color_by
+
+    @color_by.setter
+    def color_by(self, color_by: str):
+        """ set the property to color vertices by """
+        if color_by not in self.properties_to_color_by:
+            return
+        self._color_by = color_by
+        self._recolor_tracks()
+        self.events.color_by()
+        # self.refresh()
```

Comment:
If this isn't needed then we can remove it altogether?

Reply:
(same below)

Reply:
i can cut